### PR TITLE
docs: add usage with ttypescript

### PR DESCRIPTION
**Pull request by m-shaka**
Hi. I've successfully used Di-compiler with `ttypescript`. I wrote down docs about it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,6 +197,29 @@ const program = createProgram(
 program.emit(undefined, undefined, undefined, undefined, di({ program }));
 ```
 
+### Usage with ttypescript
+
+To use DI-compiler With [`ttypescript`](https://github.com/cevek/ttypescript), you need to write a transformer and set the path to `tsconfig.json`.
```
Comment (wessberg) [inline README.md:202]:
```suggestion
To use DI-compiler with [`ttypescript`](https://github.com/cevek/ttypescript), create a file that wraps the invocation of `di`:
```

File: README.md [hunk 2]
```diff
@@ -197,6 +197,29 @@ const program = createProgram(
...
+
+```typescript:transformer.ts
+import type { Program } from 'typescript'
+import { di } from "@wessberg/di-compiler";
+
+const transformer = (program: Program) => di({ program })
+
+export default transformer
+```
+
```
Comment (wessberg) [inline README.md:212]:
```suggestion
Then add a record to the `plugins` array of your `tsconfig.json` that maps a name you choose to the relative path to the file you jusit created:
```

File: README.md [hunk 3]
```diff
@@ -197,6 +197,29 @@ const program = createProgram(
...
+```tsconfig.json
+{
+    "compilerOptions": {
+        "plugins": [
+            { "transform": "path/to/transformer.ts" },
```
Comment (wessberg) [inline README.md:217]:
```suggestion
            { "di": "path/to/my-di-transformer.ts" },
```

**Review by wessberg — CHANGES_REQUESTED** [review]
Hey there. Thanks a lot for your contribution! I've added a few suggestions to your docs.

File: README.md (hunk 3 above)
Comment (m-shaka) [reply README.md:217]: It doesn't seem to work. `ttypescript`(@1.5.12) requires `"transform"` as key.

```
Error: tsconfig.json plugins error: transform must be present
```

Comment (wessberg) [reply README.md:217]: > It doesn't seem to work. `ttypescript`(@1.5.12) requires `"transform"` as key.
> 
> ```
> Error: tsconfig.json plugins error: transform must be present
> ```

Ah, OK, I didn't know that. I'll remove the suggestion

**Comment by kevinramharak** [comment]
@wessberg Can this be merged? I had to look around in the issues/PR's to find these instructions. Would make it easier for people to get started as `ttsc` is easier to setup being a drop in replacement for `tsc`.

**Comment by kevinramharak** [comment]
Thank you. Very cool package, I had the same idea last week and figured I would google first. Now im thinking of a project to use it in :).

**Pull request merged**